src/Hooks/HookManager.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Carnival\Hooks;
+
+use Carnival\Hooks\Topic;
+use Illuminate\Support\Collection;
+
+class HookManager {
+
+    protected Collection $hooks;
+
+    public function __construct() {
+        $this->hooks = collect();
+    }
+
+    function add(string $eventClass, callable $callback) : void {
```

Comment:
Just a thought: what would you think about naming this method `register` to be in line with the ExtensionManager and the other managers to come?

Reply:
Sure, to maintain consistency that's fine then.